pkg/jsonnet/importer.go

```diff
@@ -1,67 +1,122 @@
 package jsonnet
 
 import (
-	"bytes"
 	"encoding/json"
 	"io"
 	"path/filepath"
+	"strings"
 
 	jsonnet "github.com/google/go-jsonnet"
 	"github.com/pkg/errors"
 	"gopkg.in/yaml.v3"
 )
 
+const locationInternal = "<internal>"
+
 // ExtendedImporter wraps jsonnet.FileImporter to add additional functionality:
 // - `import "file.yaml"`
+// - `import "tk"`
 type ExtendedImporter struct {
-	fi *jsonnet.FileImporter
+	fi           *jsonnet.FileImporter
+	interceptors []ImportInterceptor
+	processors   []ImportProcessor
 }
 
+// ImportInterceptor are executed before the actual importing. If they return
+// something, this value is used.
+type ImportInterceptor func(importedFrom, importedPath string) (c *string, foundAt string, err error)
+
+// ImportProcessor are executed after the file import and may modify the result
+// further
+type ImportProcessor func(contents, foundAt string) (c *string, err error)
+
 // NewExtendedImporter returns a new instance of ExtendedImporter with the
 // correct jpaths set up
 func NewExtendedImporter(jpath []string) *ExtendedImporter {
 	return &ExtendedImporter{
 		fi: &jsonnet.FileImporter{
 			JPaths: jpath,
 		},
+		processors:   []ImportProcessor{yamlProcessor},
+		interceptors: []ImportInterceptor{tkInterceptor},
 	}
 }
 
 // Import implements the functionality offered by the ExtendedImporter
 func (i *ExtendedImporter) Import(importedFrom, importedPath string) (contents jsonnet.Contents, foundAt string, err error) {
+	// check if an interceptor handles this
```

Comment:
Can you help me understand this better? Is the idea that you can only have one interceptor and one postprocessor for a given import at a time? I'd like to see some stronger godocs around these interceptors and processors since the ordering is important (i.e., if two interceptors operate on yaml, only the first will ever be used).